src/worker/log.ts

```diff
@@ -25,7 +26,77 @@ export async function indexLog(block: BlockAttribute, isRetracted: boolean) {
             parseInt(block.transactionsCount, 10)
         );
     }
-    // FIXME: queryLog for each tx action
+    await getTransactionsByBlockNumber({ blockNumber: block.number }).then(
+        transactions => {
+            transactions.map(async tx => {
+                let logtype: LogType | undefined;
+                switch (tx.get().type) {
+                    case "mintAsset": {
+                        logtype = LogType.MINT_ASSET_COUNT;
+                        break;
+                    }
+                    case "transferAsset": {
+                        logtype = LogType.TRANSFER_ASSET_COUNT;
+                        break;
+                    }
+                    case "composeAsset": {
+                        logtype = LogType.COMPOSE_ASSET_COUNT;
+                        break;
+                    }
+                    case "decomposeAsset": {
+                        logtype = LogType.DECOMPOSE_ASSET_COUNT;
+                        break;
+                    }
+                    case "setRegularKey": {
+                        logtype = LogType.SET_REGULAR_KEY_COUNT;
+                        break;
+                    }
+                    case "setShardOwners": {
+                        logtype = LogType.SET_SHARD_OWNER_COUNT;
+                        break;
+                    }
+                    case "setShardUsers": {
+                        logtype = LogType.SET_SHARD_USER_COUNT;
+                        break;
+                    }
+                    case "createShard": {
+                        logtype = LogType.CREATE_SHARD_COUNT;
+                        break;
+                    }
+                    case "changeAssetScheme": {
+                        logtype = LogType.CHANGE_ASSET_SCHEME_COUNT;
+                        break;
+                    }
+                    case "increaseAssetSupply": {
+                        logtype = LogType.INCREASE_ASSET_SUPPLY_COUNT;
+                        break;
+                    }
+                    case "remove": {
+                        logtype = LogType.REMOVE_COUNT;
+                        break;
+                    }
+                    case "custom": {
+                        logtype = LogType.CUSTOM_COUNT;
+                        break;
+                    }
+                    case "wrapCCC": {
+                        logtype = LogType.WRAP_CCC_COUNT;
+                        break;
+                    }
+                    case "unwrapCCC": {
+                        logtype = LogType.UNWRAP_CCC_COUNT;
+                        break;
+                    }
+                    default:
+                        logtype = undefined;
+                        break;
+                }
+                if (logtype !== undefined) {
+                    await queryLog(isRetracted, dateString, logtype, 1);
```

Comment:
Why do you need to iterate this N times?
I think it's possible to aggregate the count for each type.